Mathlib/Algebra/Order/BigOperators/Group/Finset.lean

```diff
@@ -236,6 +236,22 @@ lemma prod_image_le_of_one_le [MulLeftMono N]
 
 end OrderedCommMonoid
 
+section ProdSum
+
+variable [CommMonoid α] [AddCommMonoid β] [Preorder β] [AddLeftMono β]
+  (s : Finset ι) {f : ι → α} (g : α → β)
+
+theorem prod_le_sum (h_one : g 1 ≤ 0) (h_mul : ∀ (a b : α), g (a * b) ≤ g a + g b) :
```

Comment:
What do you think of naming these something like `apply_prod_le_sum_apply`? I think `prod_le_sum` seems more like `(∏ x ∈ s, f x) ≤ ∑ x ∈ s, f x`.